libyul/YulName.cpp

```diff
@@ -0,0 +1,435 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+
+#include <libyul/YulName.h>
+
+#include <libyul/backends/evm/EVMDialect.h>
+
+#include <libyul/optimiser/NameCollector.h>
+
+#include <libyul/Dialect.h>
+#include <libyul/Exceptions.h>
+#include <libyul/AST.h>
+
+#include <fmt/compile.h>
+
+namespace solidity::yul
+{
+
+YulNameRepository::YulNameRepository(solidity::yul::Dialect const& _dialect):
+	m_dialect(_dialect)
+{
+	defineName("");
```

Comment:
Is this necessary? Because you are not using the return of this `defineName` call, so I'm not sure what this is supposed to do.

Reply:
This defines the special empty name. Since I know that it is happening right at the beginning after construction, it is guaranteed that the corresponding id is 0 and this allows me to make the `emptyName` function static constexpr. I'll add an assert, though, to make things clearer